File: CS/AddColumnAtRunTime/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DisplayDetailInPopupWindow.Models;

namespace DisplayDetailInPopupWindow.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }
        [HttpPost]
        public PartialViewResult GridAction() {
            return PartialView("GridPartial", CustomerRepository.GetCustomers());
        }
    }
}

File: VB/AddColumnAtRunTime/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DisplayDetailInPopupWindow.Models

Namespace DisplayDetailInPopupWindow.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Return View()
		End Function
		<HttpPost> _
		Public Function GridAction() As PartialViewResult
			Return PartialView("GridPartial", CustomerRepository.GetCustomers())
		End Function
	End Class
End Namespace
